### PR TITLE
feat(Radio): Convert radio to TypeScript

### PR DESCRIPTION
**Pull request by rebeccaalpert**
Converted to TypeScript. I made a small change to the behavior since it was triggering a React warning related to switching from an uncontrolled to a controlled component (and back again) when you clicked the controlled radio and one about defaultChecked and checked being applied at the same time.

Fixes https://github.com/patternfly/patternfly-react/issues/1997.

### CODE REVIEW

**Comment by nicolethoen** [comment]
@rebeccaalpert can you also add integration tests for this component as outlined on this [README](https://github.com/patternfly/patternfly-react/tree/master/packages/patternfly-4/react-integration)?

File: packages/patternfly-4/react-core/src/components/Radio/Radio.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import styles from '@patternfly/patternfly/components/Radio/radio.css';
+import { css, getModifier } from '@patternfly/react-styles';
+import { Omit } from '../../helpers/typeUtils';
+
+export interface RadioProps extends Omit<React.HTMLProps<HTMLInputElement>, 'label' | 'onChange'> {
```
Comment (dlabaj) [inline packages/patternfly-4/react-core/src/components/Radio/Radio.tsx]: Need to Omit type and disable here as well.  We control these internally via isDisable, and we don't want consumers changing the type of input on us.

File: packages/patternfly-4/react-core/src/components/Radio/Radio.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  'aria-label'?: string;
+}
+
+export class Radio extends React.Component<RadioProps> {
+  static defaultProps = {
+    className: '',
+    isChecked: null,
+    isDisabled: false,
+    isValid: true,
+    label: undefined,
```
Comment (dlabaj) [inline packages/patternfly-4/react-core/src/components/Radio/Radio.tsx]: I believe this is set to undefined by default so this line can be removed.

File: packages/patternfly-4/react-core/src/components/Radio/Radio.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    onChange: () => undefined
```
Comment (dlabaj) [inline packages/patternfly-4/react-core/src/components/Radio/Radio.tsx]: Same with onChanage.

File: packages/patternfly-4/react-core/src/components/Radio/Radio.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  /** A callback for when the radio selection changes. */
+  onChange?: (checked: boolean, event: React.FormEvent<HTMLInputElement>) => void;
+  /** Aria-label of the radio. */
+  'aria-label'?: string;
+}
+
+export class Radio extends React.Component<RadioProps> {
+  static defaultProps = {
+    className: '',
+    isChecked: null,
```
Comment (dlabaj) [inline packages/patternfly-4/react-core/src/components/Radio/Radio.tsx]: Shouldn't this be false?

File: packages/patternfly-4/react-core/src/components/Radio/Radio.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        <input
+          {...props}
+          aria-label={label ? undefined : ariaLabel}
+          className={css(styles.radioInput)}
+          type="radio"
+          onChange={this.handleChange}
+          aria-invalid={!isValid}
+          disabled={isDisabled}
+          checked={isChecked}
+          defaultChecked={undefined}
```
Comment (dlabaj) [inline packages/patternfly-4/react-core/src/components/Radio/Radio.tsx]: Should be false

File: packages/patternfly-4/react-core/src/components/Radio/index.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+export { Radio } from './Radio';
```
Comment (dlabaj) [inline packages/patternfly-4/react-core/src/components/Radio/index.ts]:
```suggestion
export * from './Radio';
```

File: packages/patternfly-4/react-core/src/components/Radio/Radio.tsx (hunk 5 above)
Comment (rebeccaalpert) [reply packages/patternfly-4/react-core/src/components/Radio/Radio.tsx]: It was a hacky workaround for a React warning. I figured out a better way to do it. I'm going to drop  this.

File: packages/patternfly-4/react-core/src/components/Radio/Radio.tsx (hunk 4 above)
Comment (rebeccaalpert) [reply packages/patternfly-4/react-core/src/components/Radio/Radio.tsx]: If you set it to false, it breaks the uncontrolled radio.

File: packages/patternfly-4/react-core/src/components/Radio/Radio.tsx (hunk 5 above)
Comment (rebeccaalpert) [reply packages/patternfly-4/react-core/src/components/Radio/Radio.tsx]: It's to resolve a React warning about having both checked and defaultChecked props. I can declare it as a prop and just not pass it along to the input when there's an isChecked if you'd prefer that.

File: packages/patternfly-4/react-core/src/components/Radio/Radio.tsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  isDisabled?: boolean;
+  /** Flag to show if the radio selection is valid or invalid. */
+  isValid?: boolean;
+  /** Label text of the radio. */
+  label?: React.ReactNode;
+  /** Name for group of radios */
+  name: string;
+  /** A callback for when the radio selection changes. */
+  onChange?: (checked: boolean, event: React.FormEvent<HTMLInputElement>) => void;
+  /** Aria-label of the radio. */
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/Radio/Radio.tsx]: Can you update the description here.  Can we say `Aria label` or  `The aria-label`

**Comment by rebeccaalpert** [comment]
Added demo/tests and addressed PR feedback.

File: packages/patternfly-4/react-integration/cypress/integration/radio.spec.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
+describe('Radio Demo Test', () => {
+  it('Navigate to demo section', () => {
```
Comment (tlabaj) [inline packages/patternfly-4/react-integration/cypress/integration/radio.spec.ts]: Can you add some interactive test for checking and unchecking the controlled and uncontrolled buttons?

Comment (rebeccaalpert) [reply packages/patternfly-4/react-integration/cypress/integration/radio.spec.ts]: It has one for the controlled button, but the uncontrolled buttons don't change at all HTML-wise when they're selected, so I don't know that you can write that kind of test for them.

File: packages/patternfly-4/react-core/src/components/Radio/Radio.tsx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+  name: string;
+  /** A callback for when the radio selection changes. */
+  onChange?: (checked: boolean, event: React.FormEvent<HTMLInputElement>) => void;
+  /** Aria label for the radio. */
+  'aria-label'?: string;
+}
+
+export class Radio extends React.Component<RadioProps> {
+  static defaultProps = {
+    className: '',
```
[the remark this answers is not in the document]
Comment (rebeccaalpert) [reply packages/patternfly-4/react-core/src/components/Radio/Radio.tsx]: isChecked needs to be undefined so that the uncontrolled buttons work correctly.

**Pull request merged**